src/nemos/solvers/_jaxopt_optimistix_param_mismatch_handling.py

```diff
@@ -0,0 +1,97 @@
+import inspect
+import warnings
+from typing import Any, Type
+
+
+def _clean_solver_kwargs(
+    solver_class: Type, solver_init_kwargs: dict[str, Any]
+) -> dict[str, Any]:
+    """
+    Clean up the arguments passed to the solver accounting for some mismatches between JAXopt and Optimistix.
+
+    The maximum number of iterations: JAXopt solvers use `maxiter`, `Optimistix` uses `max_steps.`
+    If the unexpected parameter name is given, just rename it in `solver_init_kwargs`.
+
+    The tolerances for the convergence criterion are defined differently.
+    JAXopt solvers typically have a single `tol` parameter, while Optimistix passes `atol` and `rtol`
+    to their Cauchy criterion.
+    If `atol` and `rtol` are passed to a JAXopt solver, `atol` is used for `tol`.
+    If `tol` is passed to an Optimistix solver, it is used to set `atol` and `rtol` is set to 0.
+
+    In all cases a warning is raise about the unexpected argument.
+    """
+    try:
+        accepted_args = solver_class.get_accepted_arguments()
+    except AttributeError:
+        # NOTE: might want to take the union with solver_class.__init__'s args?
+        accepted_args = set(inspect.getfullargspec(solver_class).args)
```

Comment:
what's going on here?

Reply:
This was a default fallback if `get_accepted_arguments` was not implemented, getting the list of arguments the solver could accept. But it's a required method now, so the whole try-except is irrelevant. I'll remove it.